pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,1356 @@
+		T::PalletId::get().into_sub_account(("ac", auction_id))
+	}
+
+	/// A helper function which checks whether an auction ending block has been reached
+	fn is_auction_ended(general_auction_data: &GeneralAuctionData<T>) -> bool {
+		<frame_system::Pallet<T>>::block_number() >= general_auction_data.end
+	}
+
+	/// A helper function which checks whether an auction is won
+	fn is_auction_won(general_auction_data: &GeneralAuctionData<T>) -> bool {
+		if !Pallet::is_auction_ended(general_auction_data) {
+			return false;
+		}
+
+		match &general_auction_data.last_bid {
+			Some(last_bid) => match general_auction_data.reserve_price {
+				Some(reserve_price) => last_bid.1 >= reserve_price,
+				None => true,
+			},
+			None => false,
+		}
+	}
+
+	fn choose_random_block_from_range(from: u32, to: u32) -> Result<u32, DispatchError> {
+		ensure!(from < to, Error::<T>::InvalidTimeConfiguration);
+		let mut random_number = Self::generate_random_number(0u32);
+
+		let difference = to.checked_sub(from).ok_or(Error::<T>::Overflow)?;
+
+		// Best effort attempt to remove bias from modulus operator.
+		for i in 1..10 {
+			if random_number < u32::MAX.saturating_sub(u32::MAX % difference) {
+				break;
+			}
+
+			random_number = Self::generate_random_number(i);
+		}
+
+		// Caution: Remainder (%) operator only safe with unsigned
+		let result = from
+			.checked_add(random_number % difference)
+			.ok_or(Error::<T>::Overflow)?;
+
+		Ok(result)
+	}
+
+	fn generate_random_number(seed: u32) -> u32 {
+		let (random_seed, _) = T::Randomness::random(&(T::PalletId::get(), seed).encode());
+		let random_number = <u32>::decode(&mut random_seed.as_ref()).unwrap_or_default();
+		random_number
+	}
+
+	fn determine_candle_closing_range(bid: &Bid<T>, auction: &CandleAuction<T>) -> Result<u32, Error<T>> {
+		let block_number: u32 = bid.block_number.try_into().map_err(|_| Error::<T>::Overflow)?;
+		let closing_start: u32 = auction
+			.specific_data
+			.closing_start
+			.try_into()
+			.map_err(|_| Error::<T>::Overflow)?;
+		let end: u32 = auction.general_data.end.try_into().map_err(|_| Error::<T>::Overflow)?;
+
+		if block_number < closing_start {
+			Ok(One::one())
+		} else if (closing_start..end).contains(&block_number) {
+			let auction_duration = end.checked_sub(closing_start).ok_or(Error::<T>::Overflow)?;
+			let block_spread = block_number.checked_sub(closing_start).ok_or(Error::<T>::Overflow)?;
+			let multiplied_block_spread = block_spread.checked_mul(100).ok_or(Error::<T>::Overflow)?;
+
+			let closing_range = multiplied_block_spread
+				.checked_div(auction_duration)
+				.ok_or(Error::<T>::Overflow)?;
+
+			Ok(if closing_range.is_zero() {
+				One::one()
+			} else {
+				closing_range
+			})
+		} else {
+			Ok(T::CandleDefaultClosingRangesCount::get())
+		}
+	}
+}
+
+///
+/// Implementation of EnglishAuction
+///
+impl<T: Config> NftAuction<T::AccountId, T::AuctionId, BalanceOf<T>, Auction<T>, Bid<T>> for EnglishAuction<T> {
```

Comment:
i would advise splitting up the auctions implementations into separate files by the type, similarly as you done with the tests